Resources/AzureDevOps/JobTemplates/deploy-environment.yml

```diff
@@ -49,122 +49,35 @@ jobs:
-        - template: ../StepTemplates/deploy-azure-function.yml
-          parameters:
-            FunctionType: 'GetAllSkillsForOccupation'
-            WebAppName: '${{ parameters.WebAppPrefix }}-stax-skillsocc-fa'
-            AzureSubscription: '${{ parameters.AzureSubscription }}'
-            SharedResourceGroup: '${{ parameters.SharedResourceGroup }}'
-            ApimServiceName: '${{ parameters.ApimServiceName }}'
-            ApimImportScript: '${{ parameters.ApimImportScript }}'
-            SpecificationPath:  '${{ parameters.SpecificationPath }}'
-            FunctionAppDomain: '${{ parameters.FunctionAppDomain }}'
-            PackageFile: '${{ parameters.PackageFile }}'
-            ResourceGroup: '${{ parameters.ResourceGroup }}'
-
-- deployment: GetOccupationsByLabel
-  pool: 'NCS - CI and CD'
-  displayName: "Deploy GetOccupationsByLabel function"
-  environment: DEV_SERVICETAXONOMY
-  dependsOn:
-  - DeployTemplate
-  - SetupConditions
-  condition: or(eq(dependencies.SetupConditions.outputs['setupConditionStep.projectsToDeploy'], '*'), contains(dependencies.SetupConditions.outputs['setupConditionStep.projectsToDeploy'], 'GetOccupationsByLabel'))
-  strategy:
-    runOnce:
-      deploy:
-        steps:
-        - task: DownloadBuildArtifacts@0
-          displayName: "Download build artifacts"
-          inputs:
-            buildType: 'current'
-            artifactName: 'DFC.ServiceTaxonomy.ApiFunction'  
-        - template: ../StepTemplates/deploy-azure-function.yml
-          parameters:
-            FunctionType: 'GetOccupationsByLabel'
-            WebAppName: '${{ parameters.WebAppPrefix }}-stax-occbylabel-fa'
-            AzureSubscription: '${{ parameters.AzureSubscription }}'
-            SharedResourceGroup: '${{ parameters.SharedResourceGroup }}'
-            ApimServiceName: '${{ parameters.ApimServiceName }}'
-            ApimImportScript: '${{ parameters.ApimImportScript }}'
-            SpecificationPath:  '${{ parameters.SpecificationPath }}'
-            FunctionAppDomain: '${{ parameters.FunctionAppDomain }}'
-            PackageFile: '${{ parameters.PackageFile }}'
-            ResourceGroup: '${{ parameters.ResourceGroup }}'
\ No newline at end of file
+- ${{ each apiPartition in parameters.Apis }}:
+  - ${{ each api in apiPartition }}:
+    - deployment: Deploy${{ api.name }}
+      displayName: "Deploy ${{ api.name }}"
+      environment: ${{ parameters.EnvironmentName }}
+      dependsOn: 
+      - DeployTemplate
+      - SetupConditions
+      - ${{ if api.dependsOn }}:
+        - Deploy${{ api.dependsOn }}
+      condition: and(succeeded(), or(eq(dependencies.SetupConditions.outputs['setupConditionStep.projectsToDeploy'], '*'), contains(dependencies.SetupConditions.outputs['setupConditionStep.projectsToDeploy'], '${{ api.name }}')))
+      strategy:
+        runOnce:
+          deploy:
+            steps:
+            - task: DownloadBuildArtifacts@0
```

Comment:
This could be removed if PublishPipelineArtifact@0 is used in the build, you can then access the artifact using $(Pipeline.Workspace)/<artifact-name>, probably a change for a future branch though